cmd/kubeadm/app/cmd/phases/controlplane.go

```diff
@@ -143,7 +143,7 @@ func getControlPlaneSubCommands(outDir, defaultKubernetesVersion string) []*cobr
 		cmd.Flags().StringVar(&cfg.KubernetesVersion, "kubernetes-version", cfg.KubernetesVersion, `Choose a specific Kubernetes version for the control plane`)
 
 		if properties.use == "all" || properties.use == "apiserver" {
-			cmd.Flags().StringVar(&cfg.API.AdvertiseAddress, "apiserver-advertise-address", cfg.API.AdvertiseAddress, "The IP address or DNS name the API server is accessible on")
+			cmd.Flags().StringVar(&cfg.API.AdvertiseAddress, "apiserver-advertise-address", cfg.API.AdvertiseAddress, "The IP address of the API server is accessible on")
```

Comment:
So I need to look through history here.  I think other changes that crib'd hostIP from self-hosting affected this.  Sorry it's taking a while but we want the original intent to be preserved.  @kad if you have a moment could you dig the change, otherwise i'll have to search for it tomorrow.

Reply:
This is actually a defect. The `kube-apiserver` only accepts an IP for `--advertise-address`.
https://kubernetes.io/docs/reference/generated/kube-apiserver/

I believe what @stevesloka has here is correct.